processor/memorylimiter/memorylimiter.go

```diff
@@ -238,44 +245,93 @@ func (ml *memoryLimiter) forcingDrop() bool {
 	return atomic.LoadInt64(&ml.forceDrop) != 0
 }
 
-func (ml *memoryLimiter) memCheck() {
+func (ml *memoryLimiter) setForcingDrop(b bool) {
+	var i int64
+	if b {
+		i = 1
+	}
+	atomic.StoreInt64(&ml.forceDrop, i)
+}
+
+func memstatToZapField(ms *runtime.MemStats) zap.Field {
+	return zap.Uint64("cur_mem_mib", ms.Alloc/1024/1024)
+}
+
+func (ml *memoryLimiter) doGCandReadMemStats() *runtime.MemStats {
+	runtime.GC()
```

Comment:
Might be useful to track as a custom metric 1) how long GC is taking to run 2) when is it running (unless I'm missing something that is already there).

Reply:
Yes. Can be added in a future PR.